src/main/java/com/softserve/teamproject/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.softserve.teamproject.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The class implements UserDetailsService interface and overrides its loadUserByUsername(String
+ * userName) method.
+ */
+@Service
+public class UserDetailsServiceImpl implements UserDetailsService {
+
+  private UserRepository userRepository;
+
+  @Autowired
+  public void setUserRepository(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  public UserDetailsServiceImpl() {
+  }
+
+  /**
+   * This is an obligatory method that implements the similar method of the
+   * UserDetailsService interface. Here we get a user from the repository by
+   * user's name, then get his role and then we return a value of UserDetails type,
+   * which is actually provided by the framework itself. It's basically the
+   * procedure of user authorization.
+   *
```

Comment:
I didn't mean to delete "parameters" and "return" description, I meant to add descriptions to them. Like:
@return returns UserDetails with user permission